terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -81,6 +88,58 @@ def update_registry_declaration(new_req, old_req, updated_content)
         end
       end
 
+      def update_lockfile_declaration(dependency)
+        @updated_lockfile_declarations ||= {}
+        @updated_lockfile_declarations[dependency] ||=
```

Comment:
Does this need to be memoized? The file updater should only have a single dependency set so should always be the same one that gets memoized

Reply:
Yeah, it's called twice, so if we don't memoize we'd perform the work twice. It's a relatively slow method as well, so seems worth to memoize.

We might be able to get rid of the `dependency` argument, but it's a bit cumbersome with how the class is set up with inheriting from base

Reply:
The guards for the lockfile in `FileUpdater#update_dependency_files` were removed including the check for whether the lockfile will update. Now the `update_lockfile_declaration` method is only called once, so I removed the memoization for the `update_lockfile_declaration` method.